packages/core/src/models/uischema.ts

```diff
@@ -284,3 +294,19 @@ export interface Categorization
    */
   elements: (Category | Categorization)[];
 }
+
+/**
+ * A union of all available UI schema elements.
+ * This includes all layout elements, control elements, label elements,
+ * group elements, category elements and categorization elements.
+ */
+export type UISchemaElement =
```

Comment:
Ah, this is nice to see! I created this same thing in jsonforms-antd-renderers, but with a type argument for the literal type of the corresponding jsonschema. Having the schema as a type argument allows us to intersect `UISchemaElement` with [SchemaAwareScope](https://github.com/great-expectations/jsonforms-antd-renderers/blob/main/src/common/schema-derived-types.ts#L54), which traverses the jsonschema (via the type system, not at runtime), to give us autocomplete and type-checking when defining UISchemas, ie for valid `scope` values, or `options` configurations.